freqtrade/freqtradebot.py

```diff
@@ -337,9 +337,7 @@ def _get_trade_stake_amount(self, pair) -> Optional[float]:
         else:
             stake_amount = self.config['stake_amount']
 
-        # TODO: should come from the wallet
-        avaliable_amount = self.exchange.get_balance(self.config['stake_currency'])
-        # avaliable_amount = self.wallets.wallets[self.config['stake_currency']].free
+        avaliable_amount = self.wallets.wallets[self.config['stake_currency']].free
```

Comment:
you sbhould add a getter for this, which returns `.free` ... 
maybe `self.wallets.get_free(self.config['stake_currency'])` ...

That getter also should mimic the current behaviour of `get_balance` - which is to return available balance for any currency requested in dry-run ...